### ISSUE TITLE
Maintaining the client

### ISSUE THREAD

**Issue by srikwit** [wisespace-io/nsqueue#2]
Hi,

Do you have any plans for maintaining this nsqueue client?
This is listed in the official list of clients at: https://nsq.io/clients/client_libraries.html

It can be migrated to support a more recent version of Rust.
Please let me know if you would need some help in doing so.

**Comment by wisespace-io** [comment]
@srikwit This client has been based on earlier version of Tokio. Since then the Tokio library has evolved a lot. It will demand some time to update but it can be done. It can be a good exercise to learn the new async features added to Rust. Help is welcome.

**Comment by srikwit** [comment]
@wisespace-io Yes, I can take this up. I have been able to migrate a previous library based on tokio-core to the current tokio crate. I will be able to push changes by this weekend. Please do help me out in case of issues.

**Comment by srikwit** [comment]
@wisespace-io I have been able to migrate most code to nightly, down to 18 errors mostly around the futures crate. Might need a few days before I can send them for your review.

**Comment by wisespace-io** [comment]
@srikwit Ok, I imagined that it would take some time.
